File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/TesisBO.java
package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.publicaciones.model.Tesis;

import java.rmi.Remote;
import java.rmi.RemoteException;

public interface TesisBO extends Remote{
    
    public Integer insertar(String titulo, String fechaPublicacion, String descripcion,
                            Integer copiasDisponibles, Integer hayFormatoFisico,Integer hayFormatoVirtual, String url, byte[] portada,
                            Integer idBibliotecaAsociada,String gradoAcademico) throws RemoteException;
    
    public Integer modificar(Integer idTesis,String titulo, String fechaPublicacion, String descripcion,
                            Integer copiasDisponibles, Integer hayFormatoFisico,Integer hayFormatoVirtual, String url, byte[] portada,
                            Integer idBibliotecaAsociada,String gradoAcademico) throws RemoteException;
    
    public Integer eliminar(Integer idTesis) throws RemoteException;
    
    public ArrayList<Tesis> listarTodos(Integer limite) throws RemoteException;
    
    public ArrayList<Tesis> listarTodos(String titulo, String fechaPublicacion, String descripcion,
                                        Integer hayFormatoFisico,Integer hayFormatoVirtual,Integer idBibliotecaAsociada,
                                        String gradoAcademico, Integer limite) throws RemoteException;
    
    public Tesis verDetallePublicacioTesis(Integer idTesis) throws RemoteException;
    
    public Integer nuevaPublicacionTesis(String titulo, String fechaPublicacion, String descripcion,
                            Integer copiasDisponibles, Integer hayFormatoFisico,Integer hayFormatoVirtual, String url, byte[] portada,
                            Integer idBibliotecaAsociada,String gradoAcademico, ArrayList<String> direccioneslocal,
                            ArrayList<Integer> autores,ArrayList<Integer> temas) throws RemoteException;
    
    public Integer modificarPublicacionTesis(Integer idTesis,String titulo, String fechaPublicacion, String descripcion,
                            Integer copiasDisponibles, Integer hayFormatoFisico,Integer hayFormatoVirtual, String url, byte[] portada,
                            Integer idBibliotecaAsociada,String gradoAcademico,
                            ArrayList<Integer> idsCopiasEjemplares, ArrayList<String> estadosCopiasEjemplares,
                            ArrayList<Integer> autores,ArrayList<Integer> temas,Integer cantCopiasPorInsertar) throws RemoteException;
    
    public ArrayList<Tesis> listarTodos_TesisDeAutorPorID(Integer idAutor) throws RemoteException;
    
    public ArrayList<Tesis> listarTodos_TesisDeAutor(String titulo, String fechaPublicacion,
                                                     String descripcion, Integer hayFormatoFisico, Integer hayFormatoVirtual,
                                                     Integer idBibliotecaAsociada, String gradoAcademico, Integer idAutor,
                                                     String nombreCompleto) throws RemoteException;
    
    public ArrayList<Tesis> listarTodos_TesisDeTemaPorID(Integer idTema) throws RemoteException;
    
    public ArrayList<Tesis> listarTodos_TesisDeTema(String ts_titulo, String fechaPublicacion,
                                                    String descripcion, Integer hayFormatoFisico, Integer hayFormatoVirtual,
                                                    Integer idBibliotecaAsociada, String gradoAcademico, Integer idTema,
                                                    String tm_titulo) throws RemoteException;
    
}

File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPublicacionBO/TesisBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;

namespace SoftBibliotecaPublicacionBO
{
    public class TesisBO:BaseBO
    {
        /*
        public int insertar(string titulo, string fechaPublicacion, string descripcion,
                            int copiasDisponibles, int hayFormatoFisico,
                            int hayFormatoVirtual, string url, byte[] portada,
                            int idBibliotecaAsociada, string gradoAcademico)
        {
            return this.WsCliente.tesis_insertar(titulo, fechaPublicacion, descripcion, copiasDisponibles,
                                                 hayFormatoFisico, hayFormatoVirtual, url, portada,
                                                 idBibliotecaAsociada, gradoAcademico);
        }

        public int modificar(int idTesis,string titulo, string fechaPublicacion, string descripcion,
                            int copiasDisponibles, int hayFormatoFisico,
                            int hayFormatoVirtual, string url, byte[] portada,
                            int idBibliotecaAsociada, string gradoAcademico)
        {
            return this.WsCliente.tesis_modificar(idTesis,titulo, fechaPublicacion, descripcion, copiasDisponibles,
                                                  hayFormatoFisico, hayFormatoVirtual, url, portada,
                                                  idBibliotecaAsociada, gradoAcademico);
        }

        public int eliminar(int idTesis)
        {
            return this.WsCliente.tesis_eliminar(idTesis);
        }

        public BindingList<tesis> listarTodos()
        {
            try
            {
                tesis[] libros = this.WsCliente.tesis_listarTodos();
                if (libros == null) return new BindingList<tesis>();
                return new BindingList<tesis>(libros);
            }
            catch (Exception ex)
            {
                return new BindingList<tesis>();
            }
        }

        public BindingList<tesis> listarTodos(string titulo, string fechaPublicacion, string descripcion,
                                              int hayFormatoFisico, int hayFormatoVirtual, int idBibliotecaAsociada,string gradoAcademico)
        {
            try
            {
                tesis[] libros = this.WsCliente.tesis_listarTodos_porParametros(titulo, fechaPublicacion, descripcion, hayFormatoFisico,
                                                                            hayFormatoVirtual, idBibliotecaAsociada, gradoAcademico);
                if (libros == null) return new BindingList<tesis>();
                return new BindingList<tesis>(libros);
            }
            catch (Exception ex)
            {
                return new BindingList<tesis>();
            }
        }


        public tesis obtenerPorId(int idTesis)
        {
            return this.WsCliente.tesis_obtenerPorId(idTesis);
        }
        */
        public tesis verDetallePublicacioTesis(int idTesis)
        {
            try
            {
                tesis unaTesis = this.WsCliente.verDetallePublicacioTesis(idTesis);
                if (unaTesis == null) unaTesis = new tesis();
                return unaTesis;
            }
            catch (Exception ex)
            {
                return new tesis();
            }
        }

        public int nuevaPublicacionTesis(string titulo, string fechaPublicacion, string descripcion,
                        int copiasDisponibles, int hayFormatoFisico, int hayFormatoVirtual, string url, byte[] portada,
                        int idBibliotecaAsociada, string gradoAcademico, string[] direccioneslocal,
                        int[] autores, int[] temas)
        {
            return this.WsCliente.nuevaPublicacionTesis(titulo, fechaPublicacion, descripcion, copiasDisponibles,
                                                        hayFormatoFisico, hayFormatoVirtual, url, portada, idBibliotecaAsociada,
                                                        gradoAcademico, direccioneslocal, autores, temas);
        }

        public int modificarPublicacionTesis(int idTesis, string titulo, string fechaPublicacion, string descripcion,
                        int copiasDisponibles, int hayFormatoFisico, int hayFormatoVirtual, string url, byte[] portada,
                        int idBibliotecaAsociada, string gradoAcademico,
                        int[] idsCopiasEjemplares, string[] estadosCopiasEjemplares,
                        int[] autores, int[] temas, int cantCopiasPorInsertar)
        {
            return this.WsCliente.modificarPublicacionTesis(idTesis, titulo, fechaPublicacion, descripcion, copiasDisponibles,
                                                            hayFormatoFisico, hayFormatoVirtual, url, portada, idBibliotecaAsociada, gradoAcademico,
                                                            idsCopiasEjemplares, estadosCopiasEjemplares, autores, temas, cantCopiasPorInsertar);
        }

        public BindingList<tesis> listarTodos_TesisDeAutorPorID(int idAutor)
        {
            try
            {
                tesis[] ltesis = this.WsCliente.listarTodos_TesisDeAutorPorID(idAutor);
                if (ltesis == null) return new BindingList<tesis>();
                return new BindingList<tesis>(ltesis);
            }
            catch (Exception ex)
            {
                return new BindingList<tesis>();
            }
        }

        public BindingList<tesis> listarTodos_TesisDeAutor(string titulo, string fechaPublicacion,
                                                 string descripcion, int hayFormatoFisico, int hayFormatoVirtual,
                                                 int idBibliotecaAsociada, string gradoAcademico, int idAutor,
                                                 string nombreCompleto)
        {
            try
            {
                tesis[] ltesis = this.WsCliente.listarTodos_TesisDeAutor(titulo, fechaPublicacion, descripcion, hayFormatoFisico, hayFormatoVirtual,
                                                                         idBibliotecaAsociada, gradoAcademico, idAutor, nombreCompleto);
                if (ltesis == null) return new BindingList<tesis>();
                return new BindingList<tesis>(ltesis);
            }
            catch (Exception ex)
            {
                return new BindingList<tesis>();
            }
        }

        public BindingList<tesis> listarTodos_TesisDeTemaPorID(int idTema)
        {
            try
            {
                tesis[] ltesis = this.WsCliente.listarTodos_TesisDeTemaPorID(idTema);
                if (ltesis == null) return new BindingList<tesis>();
                return new BindingList<tesis>(ltesis);
            }
            catch (Exception ex)
            {
                return new BindingList<tesis>();
            }
        }

        public BindingList<tesis> listarTodos_TesisDeTema(string ts_titulo, string fechaPublicacion,
                                                string descripcion, int hayFormatoFisico, int hayFormatoVirtual,
                                                int idBibliotecaAsociada, string gradoAcademico, int idTema,
                                                string tm_titulo)
        {
            try
            {
                tesis[] ltesis = this.WsCliente.listarTodos_TesisDeTema(ts_titulo, fechaPublicacion, descripcion, hayFormatoFisico, hayFormatoVirtual, idBibliotecaAsociada, gradoAcademico, idTema, tm_titulo);
                if (ltesis == null) return new BindingList<tesis>();
                return new BindingList<tesis>(ltesis);
            }
            catch (Exception ex)
            {
                return new BindingList<tesis>();
            }
        }

    }
}
